WeakAurasOptions/AuthorOptions.lua

```diff
@@ -1145,11 +1145,33 @@ local function mergeOptions(childIndex, merged, toMerge)
   end
 end
 
+local function tablesAreEqual(t1, t2)
+	if t1 == t2 then return true end
+	local ty1 = type(t1)
+	local ty2 = type(t2)
+	if ty1 ~= ty2 then return false end
+	if ty1 ~= "table" then return false end
+
+	for k1, v1 in pairs(t1) do
+		local v2 = t2[k1]
+		if v2 == nil or not tablesAreEqual(v1, v2) then return false end
+	end
+ 
+	for k2, v2 in pairs(t2) do
+		local v1 = t1[k2]
+		if v1 == nil or not tablesAreEqual(v1, v2) then return false end
+	end
+	return true
+end
+
 local function allChoicesAreDefault(data)
   for _, option in ipairs(data.authorOptions) do
-    if option.key ~= nil and option.default ~= nil and option.default ~= data.config[option.key] then
-      return false
-    end
+    if option.key ~= nil and option.default ~= nil then
```

Comment:
```diff
 if option.key ~= nil and option.default ~= nil then
-  if tablesAreEqual(option.default, data.config[option.key]) then
-  elseif option.default ~= data.config[option.key] then
+ if not tablesAreEqual(option.default, data.config[option.key]) then
    return false
  end
end
```

Reply:
Oh, yeah that's more simple.

Why is that not correct? With that, it still returns true/false as it should, no (looking at the example inputs from my latest comment)?

Reply:
Well, for one thing, you can simply check that the option class is not `noninteractive` (see the top of the file), instead of checking that the key and default aren't nil.

Reply:
Right it's indeed correct. But that's not how that check should be written. A if, whose only purpose is to act as a additional check for the elseif branch is unnecessarily complex.